vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSequencesController.java

```diff
@@ -37,46 +37,63 @@
             @PathVariable("moduleId") String moduleId, @PathVariable("spaceId") String spaceId,
-            @RequestParam(required = false, name="branchingPoint") String branchingPointId,
-            @RequestParam(required = false, name="previousSequenceId") String previousSequenceId,
-            @RequestParam(required = false, name="clearHistory") Boolean clearHistory)
-                    throws ModuleNotFoundException, SequenceNotFoundException, SlidesInSequenceNotFoundException, SpaceNotFoundException {
+            @PathVariable(name = "previewId", required = false) String previewId,
+            @RequestParam(required = false, name = "branchingPoint") String branchingPointId,
+            @RequestParam(required = false, name = "previousSequenceId") String previousSequenceId,
+            @RequestParam(required = false, name = "clearHistory") Boolean clearHistory) throws ModuleNotFoundException,
+            SequenceNotFoundException, SlidesInSequenceNotFoundException, SpaceNotFoundException {
+
         ISpace space = spaceManager.getSpace(spaceId);
         if (space == null) {
             return "redirect:/exhibit/404";
         }
         IModule module = moduleManager.getModule(moduleId);
+        model.addAttribute("module", module);
         if (module == null) {
             return "redirect:/exhibit/404";
         }
-        model.addAttribute("module", module);
         if (module.getStartSequence() == null) {
-            model.addAttribute("showAlert", true);
-            model.addAttribute("message", "Sorry, module has not been configured yet.");
-            return "/exhibition/module";
+            return moduleNotConfigured(model, previewId);
         }
-        ISequence sequenceExist=moduleManager.checkIfSequenceExists(moduleId, sequenceId);
-        if (sequenceExist==null) {
+        ISequence sequenceExist = moduleManager.checkIfSequenceExists(moduleId, sequenceId);
+        if (sequenceExist == null) {
             throw new SequenceNotFoundException(sequenceId);
         }
 
         List<ISlide> slides = sequenceManager.getSequence(sequenceId).getSlides();
         if (slides.size() == 0) {
-            model.addAttribute("showAlert", true);
-            model.addAttribute("message", "Sorry, module has not been configured yet.");
-            return "/exhibition/module";
+            return moduleNotConfigured(model, previewId);
         }
         String firstSlideId = slides.get(0).getId();
-        if(sequenceHistory.hasHistory()){
-            if(clearHistory!=null && clearHistory==true) {
-                sequenceHistory.flushFromHistory();
-            }
+        if (sequenceHistory.hasHistory() && (clearHistory != null && clearHistory == true)) {
+            sequenceHistory.flushFromHistory();
+        }
+        if (previewId != null) {
+            return String.format(
+                    "redirect:/preview/%s/%s/module/%s/sequence/%s/slide/%s?branchingPoint=%s&previousSequenceId=%s",
```

Comment:
I believe you can just use the path variables in the string and spring will take care of it, e.g.
`redirect:/preview/{previewId}/{spaceId}/...`